### ISSUE TITLE
Next OSD (2.2.0) removes Viewport.contentSize

### ISSUE THREAD

**Issue by iangilman** [msalsbery/OpenSeadragonImagingHelper#9]
See https://github.com/openseadragon/openseadragon/pull/846. 

@altert pointed out that this library uses contentSize. 

@msalsbery What do you need contentSize for? 

@avandecreme How shall we expose the info this plugin needs?

**Comment by avandecreme** [comment]
It looks like this plugin should read `source.dimensions` or `world.getBounds()`

**Comment by iangilman** [comment]
Cool. @msalsbery Can you make that change here?

**Comment by msalsbery** [comment]
@iangilman I could, but I am SO behind on the workings of OSD 2+. My imaging plugins use the native image dimensions for many purposes, but I need a serious update to the plugins since they were written for single images and all the new world stuff. I've got a bit of a catch up curve efore I can tackle it, but if someone's already in there I'd be happy to see a PR :)

**Comment by jetic83** [comment]
world.getBounds() does not exist (undefined?) but it can be world.getItemAt(0).source.dimensions

**Comment by avandecreme** [comment]
It is actually named `world.getHomeBounds()`.

**Comment by jetic83** [comment]
Oh yes, thanks, @avandecreme , thats right.

and since imaginghelper needs the image size in pixels, and getHomebounds delivers them in viewport coordinates, solution would then use `this._viewer.viewport.viewportToImageCoordinates(...)` , am I correct?

**Comment by iangilman** [comment]
`this._viewer.viewport.viewportToImageCoordinates(...)` only works properly if there's a single image. Otherwise you need to get the image you're interested in and call `viewportToImageCoordinates(...)` on that. For instance: 

```
this._viewer.world.getItemAt(0).viewportToImageCoordinates(...);
```

**Comment by iangilman** [comment]
There's now a proposed fix: https://github.com/msalsbery/OpenSeadragonImagingHelper/pull/12 Anyone want to give it a try?

**Comment by iangilman** [comment]
Fixed by #12.
